### DISCUSSION TITLE
Is it possible to fit a variogram using a general model made of multiple implemented models ?

### DISCUSSION THREAD

**Discussion by rkenko** [GeoStat-Framework/GSTools#299] [Q&A]
Hi, i wonder if one can use multiple implemented models rather than using only one model as presented in online examples. 

For instance, here is an implementation for the estimation of the empirical variogram of a field on a grid and what i want to achieve:

```
# importing modules
import matplotlib
import numpy as np
import gstools as gs
from matplotlib import pyplot as plt

# importing the image from which we seek the variogram
input_image = "my_image.png"
image            = plt.imread(input_image)[:,:,0]

# estimating the empirical variogram
y , x = np.arange(image.shape[0]), np.arange(image.shape[1])
bin_center, gamma = gs.vario_estimate((y, x), image, mesh_type="structured", bin_edges=np.arange(64))

# fitting the theorical variogram
fit_model   = gs.Gaussian(dim=2)+gs.Exponential(dim=2)+gs.Stable(dim=2)  # a not working example of what i want to achieve            
fit_model.fit_variogram(bin_center[:64], gamma[:64], nugget=False)
# printing all the parameters of the covariance model fitted
print(fit_model)                                     # i am also looking for a way to have the optimal parameters observed 
```

I am aware that it is possible to use an user defined covariance model. But i was loooking for a automatic way in order to:
 1. identify the 3 best fitting models using the procedure presented here: https://geostat-framework.readthedocs.io/projects/gstools/en/stable/examples/03_variogram/01_find_best_model.html
 2. then combine the 3 best models identified in order to have a better fit using those 3 models. 

Is it possible to do such a thing in GSTools ?

**Comment by LSchueler** [comment]
Hi rkenko,

sorry for the late reply.
I'm not sure in what you you want to combine the covariance models. In GSTools it's certainly not possible to somehow calculate an average of different covariance models, as behind the curtains, we do a lot of calculations with a given model, see e.g. [here](https://geostat-framework.readthedocs.io/projects/gstools/en/latest/examples/02_cov_model/03_spectral_methods.html#sphx-glr-examples-02-cov-model-03-spectral-methods-py).

However, some covariance models are much more flexible than others and can contain models with fewer parameters in certain limits.
The Matern model converges to the Gaussian one, see [here](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/api/gstools.covmodel.Matern.html#gstools.covmodel.Matern). So in that case, it wouldn't even make sense to combine those two.
And you can of course come up with your own models, as you already mentioned.
Maybe I can help more, if I understand what exactly you want to achieve.
